File: Leetcode/Python/_605.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        cnt = 0
        for i in range(len(flowerbed)):
            prev = 0 if i == 0 else flowerbed[i - 1]
            following = 0 if i == len(flowerbed) - 1 else flowerbed[i + 1]
            if flowerbed[i] == 0 and prev == 0 and following == 0:
                cnt += 1
                flowerbed[i] = 1
        return cnt >= n

File: Leetcode/cpp/605.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        for (int i = 0; i < flowerbed.size(); i++) {
            if (flowerbed[i] == 0) {
                bool left_side = (i == 0) || (flowerbed[i - 1] == 0);
                bool right_side = (i == flowerbed.size() - 1) || (flowerbed[i + 1] == 0);
                if (left_side && right_side) {
                    flowerbed[i] = 1;
                    n -= 1;
                    if (n == 0){
                        return true;
                    }
                }
            }
        }
        return  (n <= 0) ? true : false;
    }
};
